beacon-chain/rpc/beaconv1/validator.go

```diff
@@ -2,27 +2,165 @@ package beaconv1
 
 import (
 	"context"
-	"errors"
+	"fmt"
+	"strconv"
 
+	"github.com/pkg/errors"
+	types "github.com/prysmaticlabs/eth2-types"
 	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	iface "github.com/prysmaticlabs/prysm/beacon-chain/state/interface"
+	"github.com/prysmaticlabs/prysm/proto/migration"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
 // GetValidator returns a validator specified by state and id or public key along with status and balance.
 func (bs *Server) GetValidator(ctx context.Context, req *ethpb.StateValidatorRequest) (*ethpb.StateValidatorResponse, error) {
-	return nil, errors.New("unimplemented")
+	state, err := bs.StateFetcher.State(ctx, req.StateId)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not get state: %v", err)
+	}
+	valContainer, err := valContainerById(state, req.ValidatorId)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not get validator container: %v", err)
+	}
+	return &ethpb.StateValidatorResponse{Data: valContainer}, nil
 }
 
 // ListValidators returns filterable list of validators with their balance, status and index.
+// NOTE: missing status support.
```

Comment:
Let's open an issue to track this and change this note to a TODO(#ISSUENUM)

Reply:
We definitely need this to work